rust/lance/src/dataset/transaction.rs

```diff
@@ -423,6 +436,18 @@ impl Transaction {
                 });
                 final_indices.extend(new_indices.clone());
             }
+            Operation::ModifyIndex {
+                new_indices,
+                old_indices,
+            } => {
+                final_fragments.extend(maybe_existing_fragments?.clone());
+                final_indices.retain(|existing_index| {
+                    !old_indices
+                        .iter()
+                        .any(|old_index| old_index.uuid == existing_index.uuid)
```

Comment:
would it make more sense to call these `removed_indices` and `added_indices`?

Maybe we can roll this up into `CreateIndex` op or vice versa?

Reply:
sg.

Reply:
> Maybe we can roll this up into CreateIndex op or vice versa?

what do you mean?

Reply:
I think the argument is that we only need a single operation that is adding and/or removing indices.  A `CreateIndex` could be a `ModifyIndex` where `old_indices` is an empty vec.  This way we have fewer operations to worry over.

Reply:
We can not change `CreateIndex`'s name now, only add new fields?  

However, `CreateIndex { old_indices, new_indices }` does sound right?

Or do we want to deprecate `CreateIndex` and only have `ModifyIndex`?

Reply:
`CreateIndex { old_indices, new_indices }` or `CreateIndex { removed_indices, new_indices }` sounds good to me.